services/consensusalgo/leanhelix/algorithm.go

```diff
@@ -7,10 +7,13 @@ import (
 	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
 	"github.com/orbs-network/orbs-spec/types/go/services"
 	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"fmt"
 )
 
 func (s *service) leaderProposeNextBlockIfNeeded() error {
-	if s.blocksForRounds[s.lastCommittedBlockHeight+1] != nil {
+	nextBlockHeight := s.lastCommittedBlockHeight + 1
+
```

Comment:
suggest to change nextBlockHeight to proposedBlockHeight. 
I think we should avoid next and previous in general because they are confusing.